src/main.rs

```diff
@@ -0,0 +1,155 @@
+#![deny(unused_must_use)]
+#![deny(unsafe_code)]
+#![deny(clippy::unwrap_used)]
+
+use anyhow::Result;
+use dialoguer::{Confirm, Password, Select};
+use uom::si::thermodynamic_temperature::degree_celsius;
+use uom::si::Unit;
+
+use crate::cli::{prelude::*, Cli, Command};
+use crate::data::{Location, Provider, Weather};
+use crate::storage::Storage;
+use crate::ui::{eprintln, get_style_for_weather, println, sprintln, theme, with_spinner};
+
+mod api;
+mod cli;
+mod data;
+mod error;
+mod storage;
+mod ui;
+
+fn main() -> Result<()> {
+    human_panic::setup_panic!();
+    env_logger::init();
+    let args = Cli::parse();
+
+    match args.command {
+        Command::Configure { provider, config } => {
+            let mut storage = Storage::load(config.as_ref())?;
+            configure_provider(&mut storage, provider)?;
+            storage.store(config)?;
+        }
+        Command::Get {
+            provider,
+            location,
+            config,
+        } => {
+            let mut storage = Storage::load(config.as_ref())?;
+            let provider = choose_active_provider(&mut storage, provider);
+
+            let api_key = storage.get_api_key(provider).to_owned();
+            let api = api::new(provider, api_key);
```

Comment:
It may happen that APIs accept something more complex than just API key, and you will have to dance like `docker login` with private ECR.
It is not relevant now, but you may need to store custom data for providers in "future".

Reply:
Thank you, I will think about whether and how it can be done better.